scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py

```diff
@@ -0,0 +1,261 @@
+
+        """
+        # generate missing data if needed
+        if not self.__args.rd_uid:
+            rd_uid = self.__generate_rotating_device_uid()
+        else:
+            rd_uid = self.__args.rd_uid
+        if not self.__args.spake2_verifier:
+            spake_2_verifier = self.__generate_spake2_verifier()
+        else:
+            spake_2_verifier = self.__args.spake2_verifier
+
+        try:
+            with open(self.__args.output + "/output.json", "w+") as json_file:
+                # serialize mandatory data
+                self.__add_entry("sn", self.__args.sn)
+                self.__add_entry("date", self.__args.date)
+                self.__add_entry("hw_ver", self.__args.hw_ver)
+                self.__add_entry("hw_ver_str", self.__args.hw_ver_str)
+                self.__add_entry("rd_uid", HEX_PREFIX + rd_uid)
+                self.__add_entry("dac_cert", self.__process_der(self.__args.dac_cert))
+                self.__add_entry("dac_key", self.__process_der(self.__args.dac_key))
+                self.__add_entry("pai_cert", self.__process_der(self.__args.pai_cert))
+                self.__add_entry("cd", self.__process_der(self.__args.cd))
+                if self.__args.include_passcode:
+                    self.__add_entry("passcode", self.__args.passcode)
+                self.__add_entry("spake2_it", self.__args.spake2_it)
+                self.__add_entry("spake2_salt", HEX_PREFIX + self.__args.spake2_salt)
+                self.__add_entry("spake2_verifier", spake_2_verifier)
+                self.__add_entry("discriminator", self.__args.discriminator)
+
+                factory_data_dict = generate_dict(self.__factory_data)
+                # add user-specific data
+                factory_data_dict["user"] = self.__user_data
+
+                json_object = json.dumps(factory_data_dict)
+                json_file.write(json_object)
+
+                if self.__args.schema:
+                    self.__validate_output_json(json_object)
+                else:
+                    log.warning("Json Schema file has not been provided, the output file can be wrong. Be aware of that.")
+        except IOError as e:
+            log.error("Can not save output file into directory: {}".format(self.__args.output))
+
+    def __add_entry(self, name: str, value: any):
+        """ Add single entry to list of tuples ("key", "value") """
+        if value:
+            log.debug("Adding entry '{}' with size {} and type {}".format(name, sys.getsizeof(value), type(value)))
+            self.__factory_data.append((name, value))
+
+    def __generate_spake2_verifier(self):
+        """ If verifier has not been provided in arguments list it should be generated via external script """
+        spake2_params = gen_spake2p_params(self.__args.spake2p_path, self.__args.passcode,
+                                           self.__args.spake2_it, self.__args.spake2_salt)
+        return spake2_params["Verifier"]
```

Comment:
Like I mentioned previously, it's base64 encoded, so it must be converted to a hex string.